.github/workflows/check_new_kernel_configs.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import json
+import argparse
+import sys
+import re
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def check_kernel(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
+    if match:
+        return match.group(1)
+    else:
+        return None
+
+def check_config_arch(input_config_data):
+    if "Linux/x86_64" in input_config_data:
+        return "AMD64"
+    elif "Linux/arm64" in input_config_data:
+        return "ARM64"
+    else:
+        return None
+
+# find the lines in the diff that contain +/- CONFIG
+# ignore the lines that contain @
+# return the set of words that follow +/-
+def find_matching_lines(input_string):
+    pattern = r'(?!.*@)[+-]\s*.*CONFIG.*'
+    matching_lines = re.findall(pattern, input_string)
+    holder=[re.sub(r"\+|\-|=y|=m|\#|is not set", r"", s).strip() for s in matching_lines]
+    config_set = set(holder)
+    return config_set
+
+# parse diff for new kernel configs
+# check if they are in required configs
+def find_missing_configs(json_file, kernel, arch, config_diff):
+    # Load the JSON object
+    with open(json_file, 'r') as file:
```

Comment:
>json_file [](http://example.com/codeflow?start=14&length=9)

General comment for both scripts: when within one script we're variables representing both file paths and file objects (like the `file` variable below), I'd advise making it clear in the variable names. Example from this piece of code:
- `json_file` -> `config_file_path`
- `file` -> `config_file`.

This way it's easier to avoid mistakes and the contents of the variables are clearer to the reader.

Reply:
And I'd in general avoid generic variable names like `file`, `data`, `json`, etc. With one exception for well-established conventions within a given language like `with(...) as f`.